http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/ResponseHandler.java

```diff
@@ -74,10 +75,17 @@ protected void channelRead0(ChannelHandlerContext channelContext, HttpObject msg
         if (msg instanceof StreamedHttpResponse) {
             requestContext.handler().onStream(new PublisherAdapter((StreamedHttpResponse) msg, channelContext, requestContext));
         } else if (msg instanceof FullHttpResponse) {
-            requestContext.handler().onStream(new EmptyRequestPublisher(msg, channelContext));
-        }
-
-        if (msg instanceof LastHttpContent) {
+            // TODO: HttpStreamsClientHandler leaves a dangling LastHttpContent
+            // in the pipeline. Consume it ourselves to make sure the channel
+            // is empty at the end of stream and before releasing it back to he
+            // pool.  The HttpStreamsClientHandler should really be doing this
+            // for us.
+            channelContext.read();
+
+            ByteBuf fullContent = ((FullHttpResponse) msg).content();
+            final ByteBuffer bb = copyToByteBuffer(fullContent);
+            fullContent.release();
+            requestContext.handler().onStream(new FullResponseContentPublisher(channelContext, bb));
             Subscriber<? super ByteBuffer> subscriber = channelContext.channel().attr(ChannelAttributeKeys.SUBSCRIBER_KEY).get();
```

Comment:
Do we still need to stick the subscriber in an attribute if we complete on the same message?

Reply:
Yeah since the `FullResponseContentPublisher` is the one that sets the subscriber in the ChannelContext